packages/docs/docs/guides/5-request-client/2-erc20-payment-detection.ts

```diff
@@ -0,0 +1,211 @@
+const payerIdentity = {
+  type: RequestNetwork.Types.Identity.TYPE.ETHEREUM_ADDRESS,
+  value: '0xF317BedAA5c389F2C6f469FcF25e0752C7228Ba6',
+};
+
+/**
+ * ### Signature Provider
+ *
+ * The Ethereum private key signature provider allows a user to pass in their private Ethereum key to sign a request.
+ * The signature is a proof of who created the request and of it's integrity (that no data changed after it was signed).
+ * This process is similar to the signature of an Ethereum transaction.
+ */
+
+// The signature info requires the request creator private key.
+// Please be careful with how you store and handle your private key since it's a very sensitive piece of data.
+const payeeSignatureInfo = {
+  method: RequestNetwork.Types.Signature.METHOD.ECDSA,
+  privateKey: '0xc87509a1c067bbde78beb793e6fa76530b6382a4c0241e5e4a9ec0a0f44dc0d3',
+};
+
+const signatureProvider = new EthereumPrivateKeySignatureProvider(payeeSignatureInfo);
+
+// Add the payer private key to the signature provider to be able to accept the request
+const payerSignatureInfo = {
+  method: RequestNetwork.Types.Signature.METHOD.ECDSA,
+  privateKey: '0x05e35c0171aef18b87578800e1c15d819408df07907b6823aa48d0f568b1aa9a',
+};
+
+signatureProvider.addSignatureParameters(payerSignatureInfo);
+
+// We can initialize the RequestNetwork class with the signature provider and inform we will be using the mock storage.
+const requestNetwork = new RequestNetwork.RequestNetwork({
+  signatureProvider,
+  useMockStorage: true,
+});
+
+/**
+ * ### Request Information
+ *
+ * In the next section of code we declare the request information.
+ */
+
+// The main request info, with the currency, amount (in the smallest denominator), payee identity and payer identity
+const requestInfo: RequestNetwork.Types.IRequestInfo = {
+  currency: 'REQ',
+  expectedAmount: '1000000000000000000', // 1 REQ
+  payee: payeeIdentity,
+  payer: payerIdentity,
+};
+
+/**
+ * ## Request creation with address based payment network
+ *
+ * To create a request using the erc20 address based payment network, we need to provide the payment network parameters to the request creation parameters.
+ */
+
+const addressBasedPaymentNetwork: RequestNetwork.Types.Payment.IPaymentNetworkCreateParameters = {
+  id: RequestNetwork.Types.Payment.PAYMENT_NETWORK_ID.ERC20_ADDRESS_BASED,
+  parameters: {
+    // eslint-disable-next-line spellcheck/spell-checker
+    paymentAddress: '0x92FC764853A9A0287b7587E59aDa47165b3B2675',
+  },
+};
+
+const addressBasedCreateParams = {
+  addressBasedPaymentNetwork,
+  requestInfo,
+  signer: payeeIdentity,
+};
+
+// Finally create the request and print its id
+(async () => {
+  const request = await requestNetwork.createRequest(addressBasedCreateParams);
+
+  console.log('Request created with erc20 address based payment network:');
+  console.log(request.requestId);
+
+  // The payer can add a refund address. For example when accepting the request
+  await request.accept(payerIdentity, {
```

Comment:
confusing, not a realy life use case

we should have a separate flow for Accept
flow 1: create (and store the requestId in a global var)
flow 2: accept (fromRequestId + accept)

the 2 flow should use different instances of requestNetwork

Reply:
Stated on the last comment, I remove `accept`

Concerning `flow 1`, we can't put requestId in a global var because request creation must be done in a async function, that's why I do it for every block of code